README.adoc

```diff
@@ -0,0 +1,72 @@
+= About JICMP6
+
+_JICMP6_ is a small library to allow the use of _IPv6_ _ICMP_ (raw) packets in Java.
+
+== Build from source
+
+.Requirements
+
+* git
+* automake
+* autoconf
+* libtool
+* Java JDK 1.5+
+
+The repository has a _git_ submodule which contains _Macros_ required to compile from source code.
+
+.Clone the respository on local disk
+[source]
+----
+git clone https://github.com/OpenNMS/jicmp6.git
+----
+
+.Switch into source code repository
+[source]
+----
+cd jicmp6
+----
+
+.Initialize and update the git submodule.
+[source]
+----
+git submodule update --init --recursive
+----
+
+.Update generated configuration files with
+[source]
+----
+autoreconf -fvi
+----
+
+.Generate make files and compile JICMP6
+[source]
+----
+./configure
+make
+----
+
+.Install the library on your system, root permissions may required when working as non-root user.
+[source]
+----
+sudo make install
+----
+
+== Using JICMP6 as non-root
+
+_Mac OS X_ supports non-root _ICMP_ through the _SOCK_DRGAM_ interface, which _JICMP_ uses by default.
+
+_Linux_ supports this as well, but you additionally need to set a sysctl _OID_ to allow ping for non-root users.
+
+You can set this temporarily by running: 
+ 
+[source]
+----
+sysctl -w net.ipv4.ping_group_range="0 429496729"
```

Comment:
We should add something like: Despite having IPv4 in the option name, this also effects IPv6 sockets.